fixup! ASoC: rt711-sdca: fix IO transfer timed out when unloading the module.

shumingfan opened the pull request:
Due to the SCP_SDCA_INTMASK will be cleared by any reset,
the codec driver wants to set the setting back while the device attached again.
However, the codec driver may go to the suspend mode already at this moment.
The previous version tries to set cache_only=N and bypass the cache, then writes the registers.
There is a problem here that the cache_only feature doesn't set back to 'Y' if the codec suspended.
With this condition, the driver will try to access the registers when unloading the module.
But, the master suspended already and doesn't work for register access properly.
Therefore, this patch uses sdw_write_no_pm instead of regmap_write in the rt711_sdca_update_status().

The second patch also changed API from the regmap to sdw_write/read_no_pm.
Address 0x5c and 0x5d are related to the SDCA interrupt masks. 
Regmap should be limited to vendor-specific registers.

bardliao commented on sound/soc/codecs/rt711-sdca-sdw.c:
```diff
@@ -144,16 +144,16 @@ static int rt711_sdca_update_status(struct sdw_slave *slave,
+			sdw_write_no_pm(rt711->slave, SDW_SCP_SDCA_INTMASK1,
 				SDW_SCP_SDCA_INTMASK_SDCA_0);
-			regmap_write(rt711->regmap, SDW_SCP_SDCA_INTMASK2,
+			sdw_write_no_pm(rt711->slave, SDW_SCP_SDCA_INTMASK2,
```
We should be really careful to access a register by regmap and the IO function directly. The code looks fine to me since we set the same value here as rt711_sdca_jack_init(), and we don't read it. But I will wait for @plbossart 's comment.

plbossart replied:
indeed we should not have the INTMASK1/2 registers in regmap, so what's missing is the removal of those registers from the regmap definition. 
I think I added this in another PR some time back?

shumingfan replied:
#2622

bardliao commented:
SOFCI TEST

shumingfan commented:
@plbossart 
I modified your patch (https://github.com/thesofproject/linux/pull/2622/commits/923fec7c3117e5c0f4b54650e4aba2acfb5172f9) and added it to this PR.

bardliao reviewed (approved):
LGTM, thanks @shumingfan Should we explain why we don't use regmap functions for SDW interrupt registers in the commit message when sending these sdca codec driver upstream?

shumingfan commented:
> @plbossart
> I modified your patch ([923fec7](https://github.com/thesofproject/linux/commit/923fec7c3117e5c0f4b54650e4aba2acfb5172f9)) and added it to this PR.

@plbossart Do you have any comments?

plbossart commented:
@shumingfan I wasn't sure if you wanted this PR to be merged, the commit messages need work especially the second one that makes a reference to another commit in the SOF tree without explanations.

shumingfan commented:
> @shumingfan I wasn't sure if you wanted this PR to be merged, the commit messages need work especially the second one that makes a reference to another commit in the SOF tree without explanations.

Sorry about that. I will update the commit messages and update this PR.

shumingfan commented:
> @shumingfan I wasn't sure if you wanted this PR to be merged, the commit messages need work especially the second one that makes a reference to another commit in the SOF tree without explanations.

@plbossart I had updated the commits.

bardliao commented:
> > @shumingfan I wasn't sure if you wanted this PR to be merged, the commit messages need work especially the second one that makes a reference to another commit in the SOF tree without explanations.
> 
> @plbossart I had updated the commits.

Also, note that the SDCA codec drivers are not upstream yet. So we will squash all these fixups when sending the patch upstream. So the commit message will disappear or we have to rewrite the commit message when squashing

bardliao commented:
SOFCI TEST

bardliao reviewed (approved):
LGTM. Let's wait for the CI test result.

plbossart commented:
@bardliao should we merge? not sure what we are missing. I'd like the SDCA codecs to be submitted upstream, we are near the end of the 5.11 cycle.

bardliao commented:
The CI pr-device-test failure is NotTested and SKIP. Let's merge it.

bardliao commented:
> @bardliao should we merge? not sure what we are missing. I'd like the SDCA codecs to be submitted upstream, we are near the end of the 5.11 cycle.

@plbossart [soundwire: export sdw_write/read_no_pm functions](https://github.com/thesofproject/linux/commit/04f0534505518dd7efaca7361cbd67e667395093) is required for the sdca codec drivers. It is still not merged on upstream yet.

plbossart commented:
@bardliao can you ping Vinod and Mark on this? was there any objection?

bardliao commented:
> @bardliao can you ping Vinod and Mark on this? was there any objection?

I didn't see any objection. Vinod has pinged Mark on Jan. 6 [Re: regmap: sdw: use _no_pm functions in regmap_read/write](https://patchwork.kernel.org/project/alsa-devel/patch/20201209053459.5515-6-yung-chuan.liao@linux.intel.com/) and I resent the series on 22 Jan [[RESEND PATCH v2 0/9] soundwire/regmap: use _no_pm routines](https://lkml.org/lkml/2021/1/22/55)
